### ISSUE TITLE
Unsorted steric conflicts

### ISSUE THREAD

**Issue by tylerjereddy** [tylerjereddy/steric-conflicts#8]
Within `steric_assessment_general.main()` the returned variable (list) `cumulative_parent_list_steric_violation_counts` is not sorted because each child process can "randomly" extend this list when it happens to finish (returned residue steric conflict order is stochastic).

The downstream steric conflict resolution code uses an array built from the above list to index residues in the topology and select them for application of position restraints if they have minimal steric conflicts. In repeated rounds of the steric conflict resolution code, the randomness of the returned residue indices apparently (and not surprisingly) causes topology shuffling as the residues that are 'to be restrained' change randomly.

In short, it would be nice to write a unit test that verifies that a sorted (by index) `cumulative_parent_list_steric_violation_counts` is returned and patch the sorting behaviour into the code so that the unit test passes.
